numpy/core/tests/test_dtype.py

```diff
@@ -358,7 +358,8 @@ def test_nonequivalent_record(self):
     def test_shape_equal(self):
         """Test some data types that are equal"""
         assert_dtype_equal(np.dtype('f8'), np.dtype(('f8', tuple())))
-        assert_dtype_equal(np.dtype('f8'), np.dtype(('f8', 1)))
+        with pytest.warns(FutureWarning):
```

Comment:
Let's just fix the test here too. It's not trying to test the dtype construction here.

Reply:
what dtype equality do you want to check here?  this checks that (dtype) is equivalent to (dtype, 1) which is exactly what this PR modifies.  we can't test that (dtype, 1) is equivalent to (dtype, (1,)) given that, well, that's simply not true yet.

Reply:
You're right, my comment is misplaced here. I think I'd keep this test but with a comment explaining what the behavior will be once the deprecation expires.

Reply:
done